tests/test_star.py

```diff
@@ -314,6 +314,28 @@ def test_io():
             #np.testing.assert_almost_equal(s1.data.image.array,s2.data.image.array)
             #np.testing.assert_almost_equal(s1.data.weight.array,s2.data.weight.array)
 
+        # test read_coords_params
+        if do_params:
+            with fitsio.FITS(file_name, 'r') as fin:
+                s2coords, s2params = piff.Star.read_coords_params(fin, 'stars')
+            s2coords_manual = []
+            s2params_manual = []
+            for s in stars2:
+                s2coords_manual.append(np.array([s.data[key] for key in ['u', 'v']]))
+                s2params_manual.append(s.fit.params)
+            s2coords_manual = np.array(s2coords_manual)
+            s2params_manual = np.array(s2params_manual)
+            np.testing.assert_array_equal(s2coords, s2coords_manual)
+            np.testing.assert_array_equal(s2params, s2params_manual)
+        else:
+            # reading read_coords_params should fail with an AssertionError if there is no params
```

Comment:
This should use the np.testing.assert_raises, rather than wrapping this yourself.

Also, I don't think the underlying code should raise an AssertionError.  Rather, it should probably be a RuntimeError.  AssertionError's in primary code should indicate errors in the code itself, not in how the user used the code.